discovery-provider/plugins/notifications/src/__tests__/mappings/usdcPurchaseBuyer.test.ts

```diff
@@ -0,0 +1,149 @@
+      {
+        seller_user_id: 1,
+        buyer_user_id: 2,
+        content_type: RepostType.track,
+        content_id: 10,
+        amount: 1000
+      }
+    ])
+    await insertMobileSettings(processor.identityDB, [{ userId: 2 }])
+    await insertMobileDevices(processor.identityDB, [{ userId: 2 }])
+    await setUserEmailAndSettings(processor.identityDB, 'live', 2)
+
+    const pending = processor.listener.takePending()
+    expect(pending?.appNotifications).toHaveLength(2)
+    // Assert single pending
+    const title = 'Purchase Successful'
+    const body = `You just purchased track_title_10 from User_1!`
+    await processor.appNotificationsProcessor.process(pending.appNotifications)
+    expect(sendPushNotificationSpy).toHaveBeenCalledWith(
+      {
+        type: 'ios',
+        targetARN: 'arn:2',
+        badgeCount: 1
+      },
+      {
+        title,
+        body,
+        data: {
+          id: 'timestamp:1589373217:group_id:usdc_purchase_buyer:seller_user_id:1:buyer_user_id:2:content_id:10:content_type:track',
+          type: 'USDCPurchaseBuyer',
+          entityId: 10
+        }
+      }
+    )
+
+    expect(sendEmailNotificationSpy).toHaveBeenCalledWith({
+      userId: 2,
+      email: 'user_2@gmail.com',
+      frequency: 'live',
+      notifications: [
+        expect.objectContaining({
+          specifier: '2',
+          group_id:
+            'usdc_purchase_buyer:seller_user_id:1:buyer_user_id:2:content_id:10:content_type:track',
+          type: 'usdc_purchase_buyer'
+        })
+      ],
+      dnDb: processor.discoveryDB,
+      identityDb: processor.identityDB
+    })
+    expect(sendBrowserNotificationSpy).toHaveBeenCalledWith(
+      true,
+      expect.any(Object),
+      2,
+      title,
+      body
+    )
+  })
+
+  test('Render a single email', async () => {
+    await createUsers(processor.discoveryDB, [{ user_id: 1 }, { user_id: 2 }])
+    await createTracks(processor.discoveryDB, [{ track_id: 10, owner_id: 1 }])
+    await createUSDCPurchase(processor.discoveryDB, [
+      {
+        seller_user_id: 1,
+        buyer_user_id: 2,
+        content_type: RepostType.track,
+        content_id: 10,
+        amount: 1000
+      }
+    ])
+
+    const notifications: AppEmailNotification[] = [
+      {
+        type: 'usdc_purchase_buyer',
+        timestamp: new Date(),
+        specifier: '2',
+        group_id:
+          'usdc_purchase_buyer:seller_user_id:1:buyer_user_id:2:content_id:10:content_type:track',
+        data: {
+          buyer_user_id: 2,
+          seller_user_id: 1,
+          amount: 1000,
+          content_id: 10
+        },
+        user_ids: [2],
+        receiver_user_id: 2
+      }
+    ]
+    const notifHtml = await renderEmail({
+      userId: 2,
+      email: 'joey@audius.co',
+      frequency: 'daily',
+      notifications,
+      dnDb: processor.discoveryDB,
+      identityDb: processor.identityDB
+    })
+    expect(notifHtml).toMatchSnapshot()
```

Comment:
Wasn't aware we had tests! Awesome to see this.

Reply:
yess the plugin can be very thoroughly tested which is awesome :D